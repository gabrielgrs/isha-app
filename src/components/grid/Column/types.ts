import { ReactNode } from 'react'

export type Column = {
  children: ReactNode
  size: number

  justifySelf?: 'start' | 'center' | 'end'
  alignSelf?: 'start' | 'center' | 'end'
}
