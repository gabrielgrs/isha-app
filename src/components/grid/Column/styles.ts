import styled from 'styled-components'

import * as T from './types'

export const Wrapper = styled.div<T.Column>`
  grid-column: span ${({ size }) => size};
  justify-self: ${({ justifySelf }) => justifySelf};
  align-self: ${({ alignSelf }) => alignSelf};

  @media screen and (max-width: 992px) {
    grid-column: span 12;
  }
`
