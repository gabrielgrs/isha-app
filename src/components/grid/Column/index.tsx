import * as S from './styles'
import * as T from './types'

function Column({ children, size, justifySelf, alignSelf }: T.Column) {
  return (
    <S.Wrapper size={size} justifySelf={justifySelf} alignSelf={alignSelf}>
      {children}
    </S.Wrapper>
  )
}

export default Column
