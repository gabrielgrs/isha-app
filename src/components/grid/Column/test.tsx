import { render, screen } from 'utils/test'

import Column from './index'

describe('<Column />', () => {
  it('should render Column', () => {
    const { container } = render(<Column size={12}>I am a Column</Column>)

    expect(screen.queryByText('I am a Column')).toHaveStyle({ gridColumn: 'span 12' })
    expect(container).toMatchSnapshot()
  })
})
