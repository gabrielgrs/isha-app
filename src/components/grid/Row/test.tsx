import { render } from 'utils/test'

import Row from './index'

describe('<Row />', () => {
  it('should render Row', () => {
    const { container } = render(<Row>I am a Row</Row>)

    expect(container).toMatchSnapshot()
  })
})
