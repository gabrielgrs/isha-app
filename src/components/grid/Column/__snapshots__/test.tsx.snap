// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Column /> should render Column 1`] = `
.c0 {
  grid-column: span 12;
}

@media screen and (max-width:992px) {
  .c0 {
    grid-column: span 12;
  }
}

<div>
  <div
    class="c0"
    size="12"
  >
    I am a Column
  </div>
</div>
`;
